tests/test_playbooks/katello_manifest.yml

```diff
@@ -1,43 +1,67 @@
-  - name: "Upload a Katello manifest"
-    katello_manifest:
-      username: "{{ foreman_username }}"
-      password: "{{ foreman_password }}"
-      server_url: "{{ foreman_server_url }}"
-      validate_certs: "{{ foreman_validate_certs }}"
-      organization: "Default Organization"
-      repository_url: "https://cdn.redhat.com"
-      manifest_path: "/tmp/manifest.zip"
-      state: 'present'
-  - name: "Try to upload the same Katello manifest"
-    katello_manifest:
-      username: "{{ foreman_username }}"
-      password: "{{ foreman_password }}"
-      server_url: "{{ foreman_server_url }}"
-      validate_certs: "{{ foreman_validate_certs }}"
-      organization: "Default Organization"
-      repository_url: "https://cdn.redhat.com"
-      manifest_path: "/tmp/manifest.zip"
-      state: 'present'
-  - name: "Refresh the manifest"
-    katello_manifest:
-      username: "{{ foreman_username }}"
-      password: "{{ foreman_password }}"
-      server_url: "{{ foreman_server_url }}"
-      validate_certs: "{{ foreman_validate_certs }}"
-      organization: "Default Organization"
-      repository_url: "https://cdn.redhat.com"
-      state: 'refreshed'
+    - include: tasks/organization.yml
+      vars:
+        organization_state: present
+    - include: tasks/katello_manifest.yml
+      vars:
+        manifest_state: absent
+
+- hosts: tests
+  gather_facts: false
+  vars_files:
+    - vars/server.yml
+  tasks:
+  - name: "import manifest"
+    include_tasks: tasks/katello_manifest.yml
+    vars:
+      manifest_path: "{{ katello_manifest_path }}"
+      manifest_state: "present"
+      expected_change: true
+  - name: "import manifest again, no change"
+    include_tasks: tasks/katello_manifest.yml
+    vars:
+      manifest_path: "{{ katello_manifest_path }}"
+      manifest_state: "present"
+      expected_change: false
+  - name: "update url"
+    include_tasks: tasks/katello_manifest.yml
+    vars:
+      manifest_path: "{{ katello_manifest_path }}"
+      manifest_state: "present"
+      manifest_repository_url: "http://cdn.example.com"
+      expected_change: true
+  - name: "update url again, no change"
+    include_tasks: tasks/katello_manifest.yml
+    vars:
+      manifest_path: "{{ katello_manifest_path }}"
+      manifest_state: "present"
+      manifest_repository_url: "http://cdn.example.com"
+      expected_change: false
+# the manifest we provide in fixtures cannot be refreshed because it's been deleted from the portal
+# if you want to test manifest_state: "refreshed", you will have to provide an own manifest
+# please don't record/commit the result, as the manifest allows access to your portal account
+#  - name: "refresh manifest"
+#    include_tasks: tasks/katello_manifest.yml
+#    vars:
+#      manifest_state: "refreshed"
+#      expected_change: true
+  - name: "drop manifest"
+    include_tasks: tasks/katello_manifest.yml
+    vars:
+      manifest_state: "absent"
+      expected_change: true
+
```

Comment:
Is it worth testing that `manifest_state: "absent"` is idempotent?

Reply:
Is it idempotent?

Reply:
seems it is, added test